Src/BootCamp.Chapter/Demo.cs

```diff
@@ -0,0 +1,133 @@
+                    Console.WriteLine(_demoRunning == true ? "Demo started." : "Demo Ended.");
+                }
+            }
+        }
+
+        public Demo()
+        {
+            StartDemo();
+        }
+
+        private void StartDemo()
+        {
+            PrintMenuChoice();
+
+            OnKeyPressed += RunDemoMenu;
+
+            do
+            {
+                if (GetKeyChar() == ' ')
+                {
+                    DemoRunning = true;
+                    OnKeyPressed?.Invoke(this, null);
+                }
+            } while (DemoRunning);
+        }
+
+        private void RunDemoMenu(object sender, EventArgs eventArgs)
+        {
+            OnKeyPressed -= RunDemoMenu;
+            OnKeyPressed += PrintPeopleWithFilter;
+
+            PrintMenuChoice();
+
+            do
+            {
+                if (validChars.Contains(GetKeyChar()))
+                {
+                    OnKeyPressed?.Invoke(this, null);
+                }
+            } while (!currentChar.Equals('q'));
+
+            OnKeyPressed -= PrintPeopleWithFilter;
+        }
+
+        private void PrintMenuChoice()
+        {
+            if (DemoRunning)
+            {
+                Console.WriteLine();
+                Console.WriteLine("Press 'a' for people over 18, who do not live in UK, whose surename does not contain letter 'a'.");
+                Console.WriteLine("Press 'b' for people under 18, who do not live in UK, whose surename does not contain letter 'a'.");
+                Console.WriteLine("Press 'c' for people who do not live in UK, whose surename and name does not contain letter 'a'.");
+                Console.WriteLine("Press 'q' to exit demo.");
+                Console.WriteLine();
+            }
+            else
+            {
+                Console.WriteLine("Start demo with pressing space or any other key to quit program.");
+            }
+        }
+
+        private char GetKeyChar()
+        {
+            currentChar = Console.ReadKey(true).KeyChar;
+            return currentChar;
+        }
+
+        private void PrintPeopleWithFilter(object sender, EventArgs eventArgs)
+        {
+            var demoInstance = sender as Demo;
+            
+            switch (demoInstance.currentChar)
+            {
+                case 'a':
+                    foreach (var person in contactsCenter.Filter(PeoplePredicates.IsA))
+                    {
+                        Console.WriteLine(person);
+                    }
+
+                    PrintMenuChoice();
+                    break;
+                case 'b':
+                    foreach (var person in contactsCenter.Filter(PeoplePredicates.IsB))
+                    {
+                        Console.WriteLine(person);
+                    }
+
+                    PrintMenuChoice();
+                    break;
+                case 'c':
+                    foreach (var person in contactsCenter.Filter(PeoplePredicates.IsC))
```

Comment:
Repetitive- you can extract a function out of this and just feed a filter inside that function. That way you won't repeat 10 lines of code.

Reply:
Made a new function PrintPeople and renamed PrintPeopleFilter to ChoosePeopleFilter which use the PrintPeople function.